### PR TITLE
WIP: Add volume coupling of temperature

### PR DESCRIPTION
**Pull request by JaegerSt**
openFOAM volume coupling is implemented for the temperature field located in the AITS directory

### CODE REVIEW

**Comment by MakisH** [comment]
Thanks for the effort, @JaegerSt! This is something that people have already asked a few times for. I will have a look into it.

In the meantime, if you have any additional information/description or (more crucial) a simple test case, they would be very welcome.

**Comment by MakisH** [comment]
During the preCICE Workshop, it was mentioned that this works well, even for different scenarios. More details will probably follow soonish.

File: AITS/TemperatureAIT.C [hunk 1]
```diff
@@ -0,0 +1,82 @@
+#include "TemperatureAIT.H"
```
Comment (MakisH) [inline AITS/TemperatureAIT.C]: Together with the comment about the name `AIT`, this file does not need to have a different name.

It could either be `Temperature` in a namespace that implies volume coupling, or something like `TemperatureVolume`.

File: AITS/TemperatureAIT.C [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    (
+        &mesh.lookupObject<volScalarField>(nameT)
+    )
+    ),
+    mesh_(mesh)
+{
+    dataType_ = scalar;
+}
+
+void preciceAdapter::AITS::Temperature::write(double * buffer, bool meshConnectivity)      //Bernhard Version 2
```
Comment (MakisH) [inline AITS/TemperatureAIT.C]: I am not sure why this comment is here, but it does not look necessary.
```suggestion
void preciceAdapter::AITS::Temperature::write(double * buffer, bool meshConnectivity)
```

File: AITS/TemperatureAIT.C [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+{
+    int bufferIndex = 0;
+
+    const volScalarField & Temprtr_ = *T_;
```
Comment (MakisH) [inline AITS/TemperatureAIT.C]: Was this maybe meant to be `Tempptr`? A comment would help.

File: AITS/TemperatureAIT.H [hunk 4]
```diff
@@ -0,0 +1,50 @@
+#ifndef AITS_TEMPERATURE_H
+#define AITS_TEMPERATURE_H
+
+#include "CouplingDataUser.H"
+#include "fvCFD.H"
+
+namespace preciceAdapter
+{
+namespace AITS
```
Comment (MakisH) [inline AITS/TemperatureAIT.H]: I am not sure I understand why we need a separate module, instead of implementing the same functionality as a different writer/reader of the CHT module. I think we could save some code duplication.

File: Interface.C [hunk 5]
```diff
@@ -1,6 +1,9 @@
 #include "Interface.H"
 #include "Utilities.H"
 #include "faceTriangulation.H"
+#include <vector>
```
Comment (MakisH) [inline Interface.C]: I don't see where we are using these libraries. We could probably remove these lines.

File: Interface.C [hunk 6]
```diff
@@ -203,7 +206,69 @@ void preciceAdapter::Interface::configureMesh(const fvMesh& mesh)
...
+        numDataLocations_ = mesh.C().size();
+
+        // Count the data locations for all the patches
+        // and add the those to the previously determined number of mesh points in the volume
+        for (uint j = 0; j < patchIDs_.size(); j++)
+        {
+            numDataLocations_ +=
+                    mesh.boundaryMesh()[patchIDs_.at(j)].faceCentres().size();
+        }
+        DEBUG(adapterInfo("Number of face centres: " + std::to_string(numDataLocations_)));
```
Comment (MakisH) [inline Interface.C]: Although OpenFOAM uses British English and we check for both, we currently use US English in the log:
```suggestion
        DEBUG(adapterInfo("Number of face centers: " + std::to_string(numDataLocations_)));
```

File: MeshHandle.cpp [hunk 7]
```diff
@@ -0,0 +1,390 @@
+#include "precice/MeshHandle.hpp"
```
Comment (MakisH) [inline MeshHandle.cpp]: Why is this file here?

File: MeshHandle.hpp [hunk 8]
```diff
@@ -0,0 +1,330 @@
+#pragma once
```
Comment (MakisH) [inline MeshHandle.hpp]: Why is this file here?

**Comment by JaegerSt** [comment]
Sorry for the delayed response. This work was up to one of our students who already finished. We will hopefully manage it soon to solve the conflicts.

**Comment by MakisH** [comment]
Some news on picking up the development of volume coupling for the OpenFOAM adapter: https://github.com/precice/openfoam-adapter/pull/183#issuecomment-1125754941

**Comment by MakisH** [comment]
Closing in favor of #255, which implements all review points, offers the same features, and more. Will be there in the next release of the adapter (most probably named v1.3.0).
